controllers/api.controllers.js

```diff
@@ -53,17 +61,43 @@ exports.postNewComment = (req, res, next) => {
   const { article_id } = req.params;
   const { username, body } = req.body;
 
-  if (username === undefined) {
-    return next({ status: 400, msg: "username is required" });
+  selectArticleById(article_id)
+    .then((article) => {
+      postCommentOnArticle(article_id, username, body)
+        .then((comment) => {
+          res.status(201).send({ comment });
+        })
+        .catch((err) => {
+          next(err);
+        });
+    })
+    .catch((err) => {
+      next(err);
+    });
+};
+
+exports.addArticleVotes = (req, res, next) => {
+  const { article_id } = req.params;
+  const { inc_votes } = req.body;
+
+  if (inc_votes === undefined) {
+    return next({ status: 400, msg: "inc_votes is required" });
   }
 
-  if (body === undefined) {
-    return next({ status: 400, msg: "body is required" });
+  if (typeof inc_votes !== "number") {
+    return next({ status: 400, msg: "inc_votes must be a number" });
   }
 
-  return postCommentOnArticle(article_id, username, body)
-    .then((comment) => {
-      res.status(201).send({ comment });
+  selectArticleById(article_id)
+    .then((article) => {
+      const newVotes = article.votes + inc_votes;
```

Comment:
there is no need to query the db to get the existing votes.  you can take care of this in the SQL string with SET votes = votes + $1 (the variable passed in can just be inc_votes).